src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -172,7 +175,9 @@ def failedJobs(self, failedJobs, useMask=True):
                 workflow = job['workflow']
                 taskName = job['task']
             except KeyError as ex:
-                msg = "Missing required, non-standard key %s in job in ACDC.DataCollectionService" % (str(ex))
+                # use ex.__str__() instead of str(ex) to decouple str() from
+                # what is used in getLumiWhitelist for whiteList[str(run)]
```

Comment:
I'm not sure I understand this comment. What's the relationship of `getLumiWhitelist` with stringfying an exception?

Reply:
Ok, this is an nice topic for discussion and I have not made a decision yet.

The point is that here we use `str()` to do two different things:
1. getting a string from an exception object
2. converting `run` into something that can be hashed and used as a dictionary key

Point 2. can be problematic, since, as Stefano found out, having a mismatch between the type used to create the dictionary keys and what we compare them against can cause problems. This is true especially in python3 where bytes are not automatically encoded into unicode, for example

```plaintext
Python 3.9.0 (default, Oct 13 2020, 20:14:06) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> data = {"hello": 1}
>>> data["hello"]
1
>>> data[b"hello"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: b'hello'
```

This is why `whiteList[str(run)]` is likely to cause some that will require some more fiddling with strings in this module. However, we also use `str()` to format exception, so if we were to start fiddling with `from builtins import str` this would get different types of strings from exception and may cause to append bytes to unicode. This would happen because vanilla py2 `str(ex)` calls `ex.__str__()`, which we mapped to `__as_bytes` in WMException,  while if we use `from builtins import str` then `str(ex)` calls `ex.__unicode__()` which we mapped to `__as_unicode` in WMException. This convoluted method allows us to use `"my string" + str(ex)` in both py2 and py3, at the expense of not allowing us to use `from builtins import str; "my string" + str(ex)` in py2.

To avoid this two to usages of `str()` to affect each other in the future if we find some problems in `whiteList[str(run)]`, we can use `ex.__str__()`. 

This is me being super-paranoic. But I agree that *right now* we do not need such change, that we can postpone this to a later time, that if i can spot such a issue right now I will spot it also if and when we hape problems with `whiteList[str(run)]`. 

So I am reverting this change, always a pleasure to have a chat! Thanks for making the right questions every time!